app/controllers/users/phone_setup_controller.rb

```diff
@@ -135,5 +138,21 @@ def new_phone_form_params
         :recaptcha_mock_score,
       )
     end
+
+    def check_phone_submission_limit(phone)
+      @phone = Pii::Fingerprinter.fingerprint(Phonelib.parse(phone).e164)
+      submission_rate_limiter ||= RateLimiter.new(
+        target: @phone,
+        rate_limit_type: :phone_submissions,
+      )
```

Comment:
I think there's overlap between this and the OtpRateLimiter [here](https://github.com/18F/identity-idp/blob/1337c19d4852cb4d851f020cda1707c850b2358b/app/controllers/users/two_factor_authentication_controller.rb#L343-L349), which uses the phone number hash as the discriminator in the same way. Is it worth looking at updating that instead of creating a somewhat duplicative rate limiter?

Reply:
I was looking at OtpRateLimiter initially but since I only so far recreate the `phone_fingerprint` method. I thought maybe I'd have to do something with the `otp_findtime` and `otp_maxretry_times` functions since they are specific to how it is originally applied, but maybe they wouldn't have had any impact on using it here.
It just seemed to be less impact calling Pii::Fingerprinter.fingerprint(... than revising OtpRateLimiter.

Reply:
I guess my suggestion is more along the lines of whether we should update the configuration values used in `OtpRateLimiter` rather than add a very similar additional rate limit.

Right now, `otp_findtime` and `otp_maxretry_times` are used for both confirmed and unconfirmed phone numbers in `OtpRateLimiter`, but I think it's worth considering splitting that to have separate configurations for confirmed/unconfirmed phone numbers over the approach here.